diff --git a/app.js b/app.js
index b782a9b..2fd7c65 100644
--- a/app.js
+++ b/app.js
@@ -253,6 +253,9 @@ function startGateway (settings) {
 
   const pluginsConfig = settings.gateway ? settings.gateway.plugins : null
 
+  // always create new app
+  pluginApp = express()
+
   // load custom plugins
   if (pluginsConfig && Array.isArray(pluginsConfig)) {
     for (const plugin of pluginsConfig) {
@@ -261,7 +264,7 @@ function startGateway (settings) {
         const instance = require(plugin)({
           zwave,
           mqtt,
-          app,
+          app: pluginApp,
           logger: loggers.module(pluginName)
         })
         instance.name = pluginName
@@ -304,6 +307,11 @@ function setupInterceptor () {
 logger.info(`Version: ${utils.getVersion()}`)
 logger.info('Application path:' + utils.getPath(true))
 
+var pluginApp = undefined
+app.use(function (req, res, next) {
+  pluginApp(req, res, next)
+})
+
 // view engine setup
 app.set('views', utils.joinPath(false, 'views'))
 app.set('view engine', 'ejs')
